src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -233,10 +233,10 @@ private function processAnonymousServices(SimpleXMLElement $xml, $file)
         if (false !== $nodes = $xml->xpath('//container:argument[@type="service"][not(@id)]|//container:property[@type="service"][not(@id)]')) {
             foreach ($nodes as $node) {
                 // give it a unique name
-                $node['id'] = sprintf('%s_%d', md5($file), ++$count);
+                $node['parent'] = sprintf('%s_%d', md5($file), ++$count);
```

Comment:
If there is no id, `parent` can't collide, because the XML loader already received moved/removed those in the patched method